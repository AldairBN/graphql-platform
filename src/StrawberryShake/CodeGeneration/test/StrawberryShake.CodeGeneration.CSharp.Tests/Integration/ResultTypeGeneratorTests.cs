using System.Text;
using System.Threading.Tasks;
using Snapshooter.Xunit;
using StrawberryShake.CodeGeneration;
using StrawberryShake.CodeGeneration.CSharp;
using Xunit;

namespace StrawberryShake.Integration
{
    public class ResultTypeGeneratorTests
    {
        readonly StringBuilder _stringBuilder;
        readonly CodeWriter _codeWriter;
        readonly ResultTypeGenerator _generator;

        public ResultTypeGeneratorTests()
        {
            _stringBuilder = new StringBuilder();
            _codeWriter = new CodeWriter(_stringBuilder);
            _generator = new ResultTypeGenerator();
        }

        [Fact]
        public async Task GenerateResult_GetHeroResult()
        {
            await _generator.WriteAsync(
                _codeWriter,
                IntegrationDescriptors.CreateGetHeroResultDescriptor());

            _stringBuilder.ToString().MatchSnapshot();
        }
    }
}
