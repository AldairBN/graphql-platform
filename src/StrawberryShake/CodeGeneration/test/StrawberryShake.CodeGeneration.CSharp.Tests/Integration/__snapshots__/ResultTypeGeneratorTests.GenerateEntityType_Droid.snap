namespace StarWarsClient
{
    [global::System.CodeDom.Compiler.GeneratedCode("StrawberryShake", "11.0.0")]
    public partial class GetHeroResult
    {
        public GetHeroResult(
            IHero hero,
            string version)
        {
            Hero = hero;
            Version = version;
        }

        public IHero Hero { get; }

        public string Version { get; }
    }
}
